src/Components/WebAssembly/Build/src/Tasks/GzipCompressBlazorApplicationFiles.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Json;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Build
+{
+    public class GzipCompressBlazorApplicationFiles : Task
+    {
+        [Required]
+        public string ManifestPath { get; set; }
+
+        public override bool Execute()
+        {
+            var serializer = new DataContractJsonSerializer(typeof(ManifestData));
+
+            ManifestData manifest = null;
+            using (var tempFile = File.OpenRead(ManifestPath))
+            {
+                manifest = (ManifestData)serializer.ReadObject(tempFile);
+            }
+
+            System.Threading.Tasks.Parallel.ForEach(manifest.FilesToCompress, (file) =>
+            {
+                var inputPath = file.Source;
+                var inputSource = file.InputSource;
+                var targetCompressionPath = file.Target;
+
+                if (!File.Exists(inputSource) ||
+                    (File.Exists(targetCompressionPath) && File.GetLastWriteTime(inputSource) > File.GetLastWriteTime(targetCompressionPath)))
```

Comment:
I'm a bit suspicious of this logic as it looks like it might lead us to introduce a very subtle incrementalism bug in the near future.

I did a build with `/bl` and found that the `TargetCompressionPath` values, if you're building with the linker disabled, look like `yourProject\obj\Debug\netstandard2.1\blazor\unlinked\System.dll`. As you know, the `unlinked` directory is a workaround for a SWA SDK issue, and the intention is to eliminate this once the underlying SDK issue is fixed, simply by going back to copying directly from the underlying source and not via an intermediate location.

The problem then is that the timestamp-based incrementalism here will no longer be sufficient. When people change SDK versions or downgrade packages they depend upon, we'll then be comparing "date in nuget package dir" versus "date on compressed file". So when people downgrade packages, the check will still think their output is up-to-date even though it isn't.

Reply:
Discussed offline. I see now that the `inputSource` paths contain content hashes, so this is fine.